### ISSUE TITLE
pacman block displays 4 packages to update, but no updates are available

### ISSUE THREAD

**Issue by bluehood** [greshake/i3status-rust#490]
Hi,
my pacman block says that 4 packages can be updated, but `sudo pacman -Syu` replies that everything is up to date. I tried restarting `i3` but the number stays at 4.

I'm not sure how to debug this.
I'm using i3status-rust version 0.12.0 on Arch, installed from [the AUR](https://aur.archlinux.org/packages/i3status-rust-git/).

My pacman block configuration:

```toml
[[block]]
block = "pacman"
interval = 10
```

**Comment by ammgws** [comment]
Can you try running the command the block runs (see in https://github.com/greshake/i3status-rust/blob/master/src/blocks/pacman.rs) in a terminal and check the output?

**Comment by bluehood** [comment]
Hi,
thanks for the quick reply. If I understand the codec correctly:

```
[blue@marvin ~]$ pacman -Qu --dbpath /var/lib/pacman/
[blue@marvin ~]$ 
```

i.e. 0 lines of output

**Comment by ammgws** [comment]
How about when you run i3status-rs in the terminal? 

Output of the pacman block will be in the `full_text` part:
```
{"version": 1, "click_events": true}
[[{"background":null,"color":"#268bd2","full_text":"<span font='DejaVu Mono Sans 12'></span>","markup":"pango","separator":false,"separator_block_width":0},{"background":"#268bd2","color":"#002b36","full_text":"  13 ","markup":"pango","name":"pacman","separator":false,"separator_block_width":0}],
```

**Comment by bluehood** [comment]
```
{"background":"#268BD2FF","color":"#002B36FF","full_text":"  4 ","markup":"pango","name":"pacman","separator":false,"separator_block_width":0}
```

**Comment by ammgws** [comment]
BTW the command should be something like:

fakeroot pacman -Qu --dbpath /tmp/checkup-db-yourlinuxusername/

**Comment by bluehood** [comment]
Here's the best I could do:

```
$ mkdir /tmp/sometmpdir
$ sh -c 'fakeroot -- pacman -Sy --dbpath "/tmp/sometmpdir"'
:: Synchronizing package databases...
 core                                                                        135.6 KiB   468 KiB/s 00:00 [##############################################################] 100%
 extra                                                                      1645.2 KiB   176 KiB/s 00:09 [##############################################################] 100%
 community                                                                     4.8 MiB   553 KiB/s 00:09 [##############################################################] 100%
 multilib                                                                    164.2 KiB   444 KiB/s 00:00 [##############################################################] 100%
$ sh -c 'fakeroot pacman -Qu --dbpath "/tmp/sometmpdir/"'
$ # zero lines in output
```

**Comment by ammgws** [comment]
Does the `checkupdates-db-yourusername` dir not exist in /tmp?
```
/tmp/checkup-db-ammgws/
├── local -> /var/lib/pacman/local
└── sync
    ├── community.db
    ├── core.db
    ├── extra.db
    └── multilib.db

2 directories, 4 files
```

**Comment by bluehood** [comment]
Ah you are right, I didn't realize the directory i3status uses would be around for inspection!

```
ls checkup-db-blue/
db.lck  local@  sync/
```

That `db.lck` looked suspect so I removed it manually, but it didn't help. I also tried to remove the whole directory.
<hr>
As I was playing around with the issue, pacman finally actually did 4 upgrades (note that I'm not _that_ dumb and I did try a `pacman -Syu` a few times before reporting the issue) and the counter is now correctly reporting 0 updates.

Not sure what happened -- but feel free to close this. I will re-open the ticket in case this happens again. Sorry for the noise.
